### PR TITLE
Improve logging in e2e package

### PR DESCRIPTION
**Pull request by alexkuc**
Improve logging for GitHub CI. Example of new logging format:

- https://github.com/eventespresso/test-workflow/actions/runs/7930541082
- https://github.com/eventespresso/test-workflow/actions/runs/7930695490

Summary of changes:

- add git-related information to the job summary
- reduce the number of [annotations](https://github.com/actions/toolkit/tree/main/packages/core#annotations) to keep "noise" level to minimum

### CODE REVIEW

File: packages/e2e-tests/src/Action.ts [hunk 1]
```diff
@@ -21,11 +23,12 @@ class Action {
 		const barista = this.contexts.make('barista', this.inputs.baristaBranch());
 		const e2e = this.contexts.make('e2e', this.inputs.e2eBranch());
 		const skipTests = this.inputs.skipTests();
+		const skipBarista = this.inputs.baristaBranch().length === 0;
```
Comment (tn3rb) [inline packages/e2e-tests/src/Action.ts:26]: 👍🏻

File: packages/e2e-tests/src/Action.ts [hunk 2]
```diff
@@ -39,13 +42,33 @@ class Action {
...
+			...repos.map((repo) => {
+				return [repo.name, repo.branch, repo.commit.substring(0, 7)];
+			}),
+		]);
+		await core.summary.write();
+	}
+
 	private mkcert(): void {
-		core.info('Installing mkcert');
+		log('Installing mkcert...');
```
Comment (tn3rb) [inline packages/e2e-tests/src/Action.ts:71]: nice

File: packages/e2e-tests/src/Action.ts [hunk 3]
```diff
@@ -58,7 +81,7 @@ class Action {
 	}
 
 	private ddev(): void {
-		core.info('Installing DDEV');
+		log('Installing DDEV...');
```
Comment (tn3rb) [inline packages/e2e-tests/src/Action.ts:84]: nice

File: packages/e2e-tests/src/Cache.ts [hunk 4]
```diff
@@ -44,10 +44,10 @@ class Cache {
...
 		// generate contextual key since we are dealing with multiple repositories
 		const k = `repo-${this.repo.name}-${this.repo.branch}-${key}`;
-		if (k.length > 512) {
-			// https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action
-			const msg = `Cache key exceeded length of 512 chars: \n${k}`;
-			core.setFailed(msg);
+		const limit = 512; // https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action
+		if (k.length > limit) {
+			annotation(`Cache key exceeded length of ${limit} chars: ` + k);
+			throw new Error();
```
Comment (tn3rb) [inline packages/e2e-tests/src/Cache.ts:50]: 👍🏻

File: packages/e2e-tests/src/Yarn.ts [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 import { EnvVars } from './Action';
 import { Artifact } from './Artifact';
 import { Tar } from './Tar';
 import { GPG } from './GPG';
+import { error, log } from './utilities';
+import * as core from '@actions/core';
+import * as glob from '@actions/glob';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
```
Comment (tn3rb) [inline packages/e2e-tests/src/Yarn.ts:13]: never seen someone put global imports below local, but sure 🤷🏻

**Review by tn3rb — APPROVED** [review]
looks great 💯

**Comment by alexkuc** [comment]
Force push due to update on `main` branch

**Comment by alexkuc** [comment]
Waiting for build to finish

**Pull request merged**